### PR TITLE
Strict null checks -- 94.6%

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Raises Strict Null Checks progress to 94.6%
- 1889 of 1996 files remaining

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: src/background/deploymentUpdater.ts [hunk 1]
```diff
@@ -210,7 +210,9 @@ export async function deactivateAllDeployedMods(): Promise<void> {
 
   reportEvent(Events.DEPLOYMENT_DEACTIVATE_ALL, {
     auto: true,
-    deployments: modComponentsToDeactivate.map((x) => x._deployment.id),
+    deployments: modComponentsToDeactivate
+      .map((x) => x._deployment?.id)
+      .filter((x) => x != null),
```
Comment (BLoe) [inline src/background/deploymentUpdater.ts:215]: we use lodash `compact()` in a lot of places for this, but that's up to you, just a nit

Comment (grahamlangford) [reply src/background/deploymentUpdater.ts:215]: We used `compact` before Typescript 5.5 updated filter to work for filtering out nullish values. I think we should favor native javascript/typescript in this case

Comment (BLoe) [reply src/background/deploymentUpdater.ts:215]: Ahh ok fair enough. Sorry, I changed it here but can make a quick pr to revert that.

Comment (grahamlangford) [reply src/background/deploymentUpdater.ts:215]: What do you mean you changed it here? The PR is still open with no changes?

Comment (BLoe) [reply src/background/deploymentUpdater.ts:215]: Oh I thought I merged this on Friday haha, was looking at it on my phone earlier. I ended up just making a branch off this branch and I forgot.

Comment (grahamlangford) [reply src/background/deploymentUpdater.ts:215]: Feel free to approve and I'll merge it

**Review by BLoe — APPROVED** [review]

**Pull request merged**